api/v1alpha1/securityintentbinding_types.go

```diff
@@ -1,16 +1,17 @@
 // SPDX-License-Identifier: Apache-2.0
 // Copyright 2023 Authors of Nimbus
 
-package v1
+package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // SecurityIntentBindingSpec defines the desired state of SecurityIntentBinding
 type SecurityIntentBindingSpec struct {
-	Intents  []MatchIntent `json:"intents"`
-	Selector Selector      `json:"selector"`
+	Intents  []MatchIntent  `json:"intents"`
+	Selector MatchWorkloads `json:"selector"`
```

Comment:
How can I bind/apply a binding to more than 1 workloads? 
Previously I could achieve this by specifying labels of workloads to which I want to bind/apply binding because it was a slice.

Reply:
The selector is a set labels, and these labels are copied to by the adapters to the security engine policies - ksp, netpol, etc. Thus, we can still  specify more than one workload by specifying multiple labels

Reply:
>  when the NimbusPolicy was being genereated, we were simply copying all the labels from the any/all specifiers, and putting them in a single map on the NimbusPolicy ( ProcessMatchLabels() ).

That's a bug in SIB and NP.

But CSIB and CWNP handles multiple resources as separate resources.
https://github.com/5GSEC/nimbus/blob/51a0c5b618aefc68aa9554bae2ee32ac90e82f20/api/v1/clusternimbuspolicy_types.go#L11-L14

https://github.com/5GSEC/nimbus/blob/51a0c5b618aefc68aa9554bae2ee32ac90e82f20/api/v1/clustersecurityintentbinding_types.go#L18

For example if you specify multiple resources as follows:
```yaml
apiVersion: intent.security.nimbus.com/v1
kind: ClusterSecurityIntentBinding
...
spec:
...
  selector:
    resources:
      - kind: Pod
        name: pod1
        namespace: ns1
        matchLabels:
          app: db
          env: dev

      - kind: Pod
        name: pod1
        namespace: ns2
        matchLabels:
          app: mongo
          env: prod
```
then nimbus will generate a CWNP as follows:
```yaml
apiVersion: intent.security.nimbus.com/v1
kind: ClusterNimbusPolicy
...
spec:
...
...
  selector:
    resources:
    - kind: Pod
      matchLabels:
        app: db
        env: dev
      name: pod1
      namespace: ns1
    - kind: Pod
      matchLabels:
        app: mongo
        env: prod
      name: pod1
      namespace: ns2
```

Reply:
Two points:

 - Regarding whether it is a bug in SIB and NP - to provide support for ANY specifier is a feature to be developed, since the  security engines such as KubeArmor, NetPOl do not support the ANY feature.
 - 
 - Also, providing support in the CSIB for the ANY feature is ok from the kyverno adapter perspective since there is support in Kyverno for ANY feature.  But, when a CSIB is expanded to NimbusPolicy, then we need to make additional changes to ensure that multiple NimbusPolicies get created for each element under the ANY specifier. Or, the underlying adapter has to generate multiple policies for a given NimbusPolicy. 

My suggestion is that we go ahead with current functionality, and in case we find a need to add the ANY feature, we can do it at a later point of time.